src/averageAges.js

```diff
@@ -14,12 +14,21 @@
  *
  * @return {number}
  */
+
+function averageAge(people) {
+  return people
+    .map(person => person.died - person.born)
+    .reduce((sum, x) => sum + x)
+    / people.length;
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter(person => person.sex === 'm');
+  const menCentury = men
+    .filter(person => Math.ceil(person.died / 100) === century);
```

Comment:
Looks good, but you can just not calculate `menCentury` if there is no century passed in function. In your case you calculate `menCentury` every time, even a century was not passed in function. Maybe it's not critical, just IMHO.

Reply:
totally agree with Mykola, you should definitely fix this

Reply:
Absolutely agree! Many thanks for the help. Fixed it!